### ISSUE TITLE
[Grid - Selection] Focus indicator is not visible on checkboxes inside grid in Safari

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3354]
### 😯 Current Behavior

When keyboard focus is placed on checkboxes inside Grid focus indicator is not visible.
This happens only in Safari.

https://github.com/user-attachments/assets/13d74011-0e00-4f4e-887f-50731501d8af

### 🤔 Expected Behavior/possible solution?

Focus indicator should be visible as blue rectangle as it is in other browsers.

https://github.com/user-attachments/assets/9139793c-6a35-4064-a863-7333a0bd6346

### 🖥️ Steps to Reproduce

1. In Safari go to: https://storybook.sanomalearning.design/?path=/story/grid-selection--multiple
2. Using `Tab` key place focus on checkbox 
3. Check if focus indicator is visible on focused checkbox

### Were you able to reproduce this in our storybook?

Yes

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Safari

### Component version

v0.7.2

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
